### ISSUE TITLE
Создание Черновика Встречи

### ISSUE THREAD

**Issue by kulakovt** [DotNetRu/Server#13]
Необходимо дать пользователю возможность создавать и изменять Черновик предстоящей Встречи.

## Понятие «Черновика»

Перед тем, как Встреча будет опубликована для регистрации, она прибывает в состоянии «черновика». В этом статусе она подготавливается для всеобщего обозрения и может содержать не консистентные данные, рабочие поля, несуществующие ссылки. «Черновик Встречи» является источником для создания в будущем «Встречи», которая будет опубликована в анонсе и сохранена в Аудите.

## Состав

Черновик должен состоять из тех же полей что и основная Встреча (см. [формат Аудита](https://github.com/DotNetRu/Audit)). Черновик встречи предполагает возможность включения в себя также и Черновиков вложенных сущностей (Доклад, Докладчик, Площадка, Друзья). В процессе редактирования вложенных сущностей должна быть возможность как сослаться на существующий экземпляр так и создать новый Черновик сущности.

## Редактирование

Сервер должен предоставлять API, который позволит не только создать встречу, но и отредактировать её по частям. Т.е. информация о будущих выступлениях будет появляться не сразу. Вполне вероятен сценарий в котором, сначала появляются Докладчики, через некоторое время у них формируются Доклады, после чего мы находим Площадку для выступления и назначаем дату Встречи. Между всеми этими шагами Сервер должен обеспечивать сохранность промежуточных данных.

> :question: Необходимо придумать удобную систему REST ресурсов для работы с иерархией встреч

## Чтение

Сервер должен предоставлять API для чтения списка Черновиков и получения информации по всему их дереву вложенных сущностей.

> :question: Как клиент должен контролировать глубину запрашиваемого дерева?

## Авторизация и аутентификация

На данном этапе развития, необходимо уделить _минимальное_ внимание этому вопросу. Например создать для всех посвящённых один пароль или сгенерировать уникального пользователя каждому кто первый раз посещает редактор.

> :question: Как меньше всего задумываться о безопасности на начальном этапе?

## Взаимодействие с пользователем

Для взаимодействия с пользователем будет использоваться [проект WebUI](https://github.com/DotNetRu/WebUI). Подробности про заведение нового черновика встречи можно почитать в DotNetRu/WebUI#1.

Кроме заведения, необходимо дать доступ ко всем черновикам для их редактирования. А также должна быть возможность удаления не нужного черновика.

## Сценарий использования Черновика

1. Я, как лидер сообщества, нахожу докладчика и мы договариваемся о подготовке выступления.
1. В этот момент я завожу «Черновик Доклада». При создании документа у него уже должны быть обязательно известны _информация о спикере_ и _тема доклада_.
1. Далее происходит серия тренировок Доклада. В процессе которых мы формируем содержимое и готовим слайды.
1. После достижения приемлемого качества Доклада, я начинаю планировать новую Встречу.
1. Для этого мне необходимо найти Площадку и договориться с ней о дате проведения мероприятия.
1. Также найти Друзей и обсудить подарки на вечер.
1. Из «дотренированных» докладов собираю программу. Прошу от Докладчиков сообщить описание выступления и прислать фотографии.
1. Заполняю все оставшиеся поля, необходимые для публикации анонса Встречи. После публикации жизнь черновика заканчивается, а все сущности переносятся в «чистовик» (Аудит).

**Comment by kulakovt** [comment]
Добавил раздел «Сценарий использования Черновика».  @a-merson @Sergey-Buyanov , зацените.

**Comment by Sergey-Buyanov** [comment]
@AnatolyKulakov с мой стороны, норм. В будущем,  в принципе можно сделать, чтобы спикер мог сам редактировать свой черновик доклада (выкладывать слайды, содержимое) в системе

**Comment by kulakovt** [comment]
@Sergey-Buyanov , да, это очень хорошая идея. Можно даже дать доступ к редактированию Friends. Но всё это ~в светлом будущем~ после реализации аутентификации.
